test/testutil.c

```diff
@@ -145,19 +145,110 @@ int run_tests(const char *test_prog_name)
     return EXIT_SUCCESS;
 }
 
+void fail_message(const char *desc, va_list args, const char *type,
```

Comment:
I don't see how adding FILE & LINE to the arguments is going to be possible.  We'd want macros to do this which then call the underlying functions with the extra arguments.  How do I generate the macros automatically?  Generating the functions is easy enough and doing macros for predefined ones would be possible (by listing them out).  How would I go with e.g. IMPLEMENT_COMPARISONS(struct fred *, fredptrs, "%p")
I'd need to generate fredpts_eq(...) and the wrapper macro TEST_fredpts_eq(...).